### ISSUE TITLE
URL parameters:  The regular expression question mark is superfluous

### ISSUE THREAD

**Issue by katongzhong1** [Chalarangelo/30-seconds-of-code#95]
I tested, the correct string is ([^?=&]+)(=([^&]*)), no ([^?=&]+)(=([^&]*))?
 Am i wrong?

**Comment by katongzhong1** [comment]
Never mind! It has changed the way just now

**Comment by katongzhong1** [comment]
Sorry, I has tried the new way, the same Problem is founded. The regular expression question mark is superfluous

**Comment by skatcat31** [comment]
With current code:

```js
const getUrlParameters = url =>
  url.match(/([^?=&]+)(=([^&]*))?/g).reduce(
    (a, v) => (a[v.slice(0, v.indexOf('='))] = v.slice(v.indexOf('=') + 1), a), {}
  );
undefined
getUrlParameters('http://url.com/page?name=Adam&surname=Smith')
var result = {"http://url.com/pag": "http://url.com/page", "name": "Adam", "surname": "Smith"}
```

Suggested change:
```js
const getUrlParameters = url =>
  url.match(/([^?=&]+)(=([^&]*))/g).reduce(
    (a, v) => (a[v.slice(0, v.indexOf('='))] = v.slice(v.indexOf('=') + 1), a), {}
  );

```

**Comment by Chalarangelo** [comment]
Resolved in #96
